WordPress/gutenberg
# Spacing support: Add server-side support for margins

## Description
This PR adds the margin block support on the server-side allowing dynamic blocks to use it. The existing `padding.php` block support file is renamed to `spacing` to reflect the existing block support's structure.

## How has this been tested?



Testing instructions
Your theme will need to opt into padding support. You can do this by updating your `theme.json` to enable padding under the spacing control, in either your default settings or the block context for button:
```
"settings": {
    "spacing": {
        "customPadding": true,
        "customMargin": true,
        "units": [
            "px",
            "em",
            "rem",
            "vh",
            "vw"
        ]
    }
}
```
I used the Site Title block for testing, as it supports both margin and padding on trunk.

**Test Instructions**

You can verify the bug first on trunk by:
1. Create a post and add a Site Title block.
2. In the Inspector Controls, give the block margin _and_ padding.
3. Save and view on the frontend. You should see that the padding has been applied, but the margin has not. (Inspect the element in the console to easily verify that the margin styles have not been added).

Test the PR:
1. Create a post, add a Site Title block.
2. In the Inspector Controls, give the block margin _and_ padding.
3. Save and view on the frontend. You should now see that both the margin and padding have been applied, and can verify by inspecting the element.

**Note that if added as a top-level block, left and right margins may be overridden by the `!important` styles from the layout block support. You should still be able to see the styles added.


## Types of changes
Bug fix (non-breaking change which fixes an issue)



## Checklist:
